Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM python:3.10.5-buster as base
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH "$PATH:/root/.local/bin/"
+EXPOSE 5000
+RUN mkdir /app
+WORKDIR /app
+COPY poetry.lock poetry.toml pyproject.toml ./
+RUN poetry install
+COPY ./todo_app ./todo_app
+
+FROM base as production
+ENTRYPOINT ["poetry", "run", "gunicorn",  "-b", "0.0.0.0:5000", "todo_app.app:create_app()"]
+
+FROM base as development
+ENTRYPOINT ["poetry", "run", "flask", "run", "-h", "0.0.0.0", "-p", "5000"]
```

Comment:
This is fine, though there are other things you could do to distinguish the two stages.

* If your dependencies in pyproject.toml were split between proper dependencies and the "dev-dependencies" section, you could skip the dev ones with `poetry install --no-dev`
* You could set some environment variables e.g. FLASK_ENV=production inside the Dockerfile, though it gets overridden by docker run --env-file or -e.